stdlib/mexpr/pprint.mc

```diff
@@ -1,51 +1,65 @@
--- TODO Update and organize according to changes in ast.mc
 
 include "char.mc"
 include "option.mc"
 include "seq.mc"
 include "string.mc"
+include "name.mc"
+include "symbol.mc"
 
-include "ast.mc"
-include "ast-builder.mc"
+include "mexpr/ast.mc"
+include "mexpr/ast-builder.mc"
 
 let spacing = lam indent. makeSeq indent ' '
 let newline = lam indent. concat "\n" (spacing indent)
 
 -- Set spacing on increment
-let incr = lam indent. addi indent 4
+let incr = lam indent. addi indent 2
+
+let symbolDelim = "'"
 
 -- Constructor name translation
-let conName = lam name.
-  if eqi (length name) 0 then
-    "#con\"\""
-  else if is_upper_alpha (head name) then
-    name
-  else
-    strJoin "" ["#con\"", name, "\""]
-
-let varName = lam name.
-    if eqi (length name) 0 then
-      "#var\"\""
-    else if is_lower_alpha (head name) then
-      name
-    else
-      strJoin "" ["#var\"", name, "\""]
+let conString = lam name.
+  match name with (str,sym) then
+    let str =
+      if eqi (length str) 0 then
+        "#con\"\""
+      else if is_upper_alpha (head str) then
+        str
+      else
+        join ["#con\"", str, "\""] in
+    let sym = if nameHasSym name then sym2string sym else "" in
+    strJoin symbolDelim [str, sym]
```

Comment:
We need to discuss how to pretty print this. The main idea of the name module was not to expose the actual symbol when pretty printing. Instead, the actual string should be used. But, the pretty printer must keep an environment which makes sure that the string is locally unique, potentially adding a number to the name, which makes it unique. By doing so, we get a much more readable pretty printer that always creates correct program code (that can be compiled).

Hence, for pretty printing names of terms, you do not need sym2string. Note the special case if you pretty print open terms (no lambda or match constract that is catching the variable name). In such a case, we should probably use the real name, but append/append it with some special string. E.g., if we have a name "x" we could pretty print it is "open_x" or something similar.

Reply:
Improving the pretty printer in this way is a non-trivial task, which is why I did not attempt it in this PR. I agree that we should do this in a future PR (in which we can also remove/change sym2int and sym2string).

I think it is best to merge this PR now, however, so that other people can start incorporating the new symbol stuff in their implementations. I can take a look at improving the pretty printing afterwards (it is actually already in my TODO list).